cli/src/lib.rs

```diff
@@ -149,23 +151,29 @@ impl NodeBuilder {
         self
     }
 
+    /// Adds a service instance that will be available immediately after creating a genesis block.
+    ///
+    /// Make sure that artifact for this instance is provided to this builder.
+    pub fn with_default_instance(mut self, instance: impl Into<InstanceInitParams>) -> Self {
+        self.default_instances.push(instance.into());
+        self
+    }
+
+    /// Adds a default Rust service instance that will be available immediately after creating a
+    /// genesis block.
+    pub fn with_default_rust_service(self, service: impl DefaultInstance) -> Self {
```

Comment:
I'd suggest to add a couple of tests to check that the builder performs decoupling as necessary; e.g., if `with_default_rust_service` and `with_default_instance` are called for the same service factory.